packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -253,21 +275,239 @@ task("setup-l2-xchain", "Configures L2 cross chain smart contracts").setAction(a
+    const [oracleHubRegistered] = await Promise.all([
+      l1RegistryContract.methods.isContractRegistered(oracleHub.address).call(),
+    ]);
+    console.log(`- OracleHub registered: ${oracleHubRegistered ? "✅" : "❌"}`);
+    console.groupEnd();
+
+    console.groupEnd();
+    /** ***********************************
+     * End: Checking L1 State
+     *************************************/
+
+    /** ***********************************
+     * Begin: Checking L2 State
+     *************************************/
+    console.group("\n🌚 Verifying L2 contract state 🌚");
+
+    console.group("Registry");
+    const l2RegistryContract = new l2Web3.eth.Contract(l2Registry.abi, l2Registry.address);
+    const [optimisticOracleRegistered, l2RegistryOwner] = await Promise.all([
+      l2RegistryContract.methods.isContractRegistered(l2OptimisticOracle.address).call(),
+      l2RegistryContract.methods.getMember(0).call(),
+    ]);
+    console.log(`- OptimisticOracle registered: ${optimisticOracleRegistered ? "✅" : "❌"}`);
+    console.log(`- Owned by GovernorSpoke: ${l2RegistryOwner === governorSpoke.address ? "✅" : "❌"}`);
+    console.groupEnd();
+
+    console.group("Store");
+    const l2StoreContract = new l2Web3.eth.Contract(l2Store.abi, l2Store.address);
+    const [l2StoreOwner] = await Promise.all([l2StoreContract.methods.getMember(0).call()]);
+    console.log(`- Owned by GovernorSpoke: ${l2StoreOwner === governorSpoke.address ? "✅" : "❌"}`);
+    console.groupEnd();
+
+    console.group("IdentifierWhitelist");
+    const l2IdentifierWhitelistContract = new l2Web3.eth.Contract(
+      l2IdentifierWhitelist.abi,
+      l2IdentifierWhitelist.address
+    );
+    const [l2IdentifierWhitelistOwner] = await Promise.all([l2IdentifierWhitelistContract.methods.owner().call()]);
+    console.log(`- Owned by GovernorSpoke: ${l2IdentifierWhitelistOwner === governorSpoke.address ? "✅" : "❌"}`);
+    console.groupEnd();
+
+    console.group("AddressWhitelist");
+    const l2AddressWhitelistContract = new l2Web3.eth.Contract(l2AddressWhitelist.abi, l2AddressWhitelist.address);
+    const [l2AddressWhitelistOwner] = await Promise.all([l2AddressWhitelistContract.methods.owner().call()]);
+    console.log(`- Owned by GovernorSpoke: ${l2AddressWhitelistOwner === governorSpoke.address ? "✅" : "❌"}`);
+    console.groupEnd();
+
+    console.group(`${l2ChainName}_ChildMessenger`);
+    const childMessengerContract = new l2Web3.eth.Contract(childMessenger.abi, childMessenger.address);
+    const [childMessengerOracleSpoke, childMessengerParentMessenger] = await Promise.all([
+      childMessengerContract.methods.oracleSpoke().call(),
+      childMessengerContract.methods.parentMessenger().call(),
+    ]);
+    console.log(`- Set oracleSpoke: ${childMessengerOracleSpoke === oracleSpoke.address ? "✅" : "❌"}`);
+    console.log(`- Set parentMessenger: ${childMessengerParentMessenger === parentMessenger.address ? "✅" : "❌"}`);
```

Comment:
This name is super confusing, but I really don't have anything better in mind 😓

Reply:
yeah we use this exact name in the contracts too so at least its consistent